crates/nu-parser/src/parse_keywords.rs

```diff
@@ -334,6 +334,7 @@ pub fn parse_for(working_set: &mut StateWorkingSet, lite_command: &LiteCommand)
     let var_type = match iteration_expr_ty {
         Type::List(x) => *x,
         Type::Table(x) => Type::Record(x),
+        Type::Range => Type::Any, // Range elements can be int or float
```

Comment:
Not sure if there's a better approach here, this made the test case pass
```suggestion
        Type::Range => Type::Number, // Range elements can be int or float
```